### ISSUE TITLE
[Feature Request] Notify users about their relations being severed by defederation

### ISSUE THREAD

**Issue by inga-lovinde** [tootsuite/mastodon#15891]
### Pitch

Consider instance X with users A@X, B@X, C@X and instance Y with users K@Y, L@Y, M@Y, with the following follow relationships: A@X<->K@Y, A@X->L@Y, C@X->K@Y, K@Y->B@X, M@Y->A@X.

When moderators of instance X suspend instance Y, users should receive the following notifications:

* A@X should receive a notification "moderators of your instance decided to suspend instance Y, which means that your mutual relationship with K@Y is removed, you can no longer follow L@Y, and you are no longer followed by M@Y".
* K@Y should receive a notification "instance Y defederated from your instance, which means that your mutual relationship with A@X is removed, you can no longer follow B@X, and you are no longer followed by C@X".
* B@X, C@X, L@Y, M@Y should receive some similar notifications.

### Motivation

It is not unusual when moderators of one medium-sized (100-500 active users) instance decide to suspend another medium-sized instance for some reason.

However, it is completely not transparent to users, who just lose their connections instantly, and might even not notice that for some time.

Consider Alice on X being friends with Bob on Y, but not talking much outside the fediverse. Now let's say administrators of Y have some really bad practices, so X decides to unfederate from them.

For Bob it just looks like Alice stopped posting. After a while Bob might remember that he didn't see Alice post for a while, worry if anything happened to Alice, try to find her, but realize she's not on his lists anymore. Maybe, if Bob will remember Alice's exact username, he'll be able to find her in Search UI just to see that he's not following her anymore. The obvious conclusion would be that Alice softblocked Bob, and Bob, respecting her borders, will not try to contact her again.

For Alice, it could be a similar experience, unless she actively follows the moderation deliberations (if there are any), is aware that moderators of X are going to suspend Y, and is aware that Bob is on Y. With the one exception: she won't be able to find Bob in Search UI. 

With a proposed change, both Alice and Bob will be aware of what's happening, will know that their connection was removed because of the moderators' actions, will have an option to decide to make an alt on another instance (or to use an existing alt) to reconnect, and their friendship could continue.

It will also make actions taken by moderators somewhat more transparent to the users of their instance.

**Comment by ClearlyClaire** [comment]
I don't think notifying users of the **blocked** instance is wise in general.

However, notifying affected users on the local instance probably makes sense, yes, although I'm not too sure in which form we should do that.

**Comment by Gargron** [comment]
I think it’s a good idea to generate a notification when a follow relationship is removed through such a block. 

Maybe the notification itself could refer to multiple accounts at once to avoid spamming the notifications column.

**Comment by whetstone303** [comment]
I’m a new fediverse user and think this feature is very important to ensure that users are aware of how the actions/non-actions of their instance moderators impact their social network. Choosing an instance that aligns with your personal preferences is only possible if moderator decisions are transparent.

I’m not a programmer but I really hope someone takes this on!

**Comment by rghedin** [comment]
Up? This is such a great feature and I'm surprised not to find it on Mastodon public roadmap.

It would save me some trouble recently, [as described in this blog post](https://notes.ghed.in/posts/2023/moderating-instances-mastodon-blocking/).

**Comment by rscmbbng** [comment]
Not only does it allow users to restore connections, but in cases such as the above, it also provides information for users that can help keep the admin team accountable.

**Comment by Stevoisiak** [comment]
Notifications for lost connections is especially important with [servers discussing defederation if Instagram Threads implements ActivityPub support](https://www.wired.com/story/metas-threads-could-make-or-break-the-fediverse/). If implementation does happen, I would expect an increase in defederations across the board.

**Comment by ShadowJonathan** [comment]
I think https://github.com/mastodon/mastodon/issues/25526 is related to this

**Comment by ShadowJonathan** [comment]
https://github.com/mastodon/mastodon/pull/27511 just merged, but iirc that only shows notifications on the server that is *doing* the defed, right? (cc @ClearlyClaire)

**Comment by renchap** [comment]
Yes, #27511 is to inform local users when their admin block a remote instance. There is nothing planned to inform a remote instance when they are blocked, as this might not be a desirable behaviour (for bad instances).

**Comment by inga-lovinde** [comment]
I'd say that, while it might not be a desirable behavior to inform the instance with followers, it might still make sense to inform the instance with mutual (or maybe even following) relationships.
And in most cases there probably will not be any mutual relationships with bad instances anyway.
